### ISSUE TITLE
Python custom op registration doesn't broadcast optional int arguments

### ISSUE THREAD

**Issue by IvanYashchuk** [pytorch/pytorch#83506]
### 🐛 Describe the bug

Try the following code. `dim` argument when passed a single integer should internally be expanded to a list of size 1 with given int. It doesn't work when the argument is specified to be optional and works as expected without `?`.

```py
import torch

test = torch.library.Library("test", "DEF")
test_impl = torch.library.Library("test", "IMPL", "CompositeExplicitAutograd")

schema = "test(int[1]? dim) -> Tensor"

test.define(schema)
test_impl.impl("test", lambda dim=None: torch.empty(dim))

try:
    torch.ops.test.test(dim=2)
except RuntimeError as e:
    print(e)

# But the following works:
import torch

schema = "test_non_optional(int[1] dim) -> Tensor"

test.define(schema)
test_impl.impl("test_non_optional", lambda dim=None: torch.empty(dim))

try:
    torch.ops.test.test_non_optional(dim=2)
except RuntimeError as e:
    print(e)
```

### Versions

Latest master.
